rasa/nlu/featurizers/sparse_featurizer/count_vectors_featurizer.py

```diff
@@ -391,9 +392,22 @@ def _merge_new_vocabulary_tokens(
         Args:
             existing_vocabulary: existing vocabulary
             vocabulary: set of new tokens
+
+        Raises:
+            RasaException: if `use_shared_vocab` is set to True and there are new
+                           vocabulary items added during incremental training.
         """
         for token in vocabulary:
             if token not in existing_vocabulary:
+                if self.use_shared_vocab:
+                    raise RasaException(
+                        "Using a shared vocabulary is not supported during "
+                        "incremental training. It requires dynamically adjusting "
+                        "layers that correspond to label attributes such as "
+                        "INTENT_RESPONSE_KEY, INTENT, etc. This is currently not "
+                        "possible. In order to avoid this exception we suggest to "
+                        "set `use_shared_vocab` to False."
```

Comment:
f"Using a shared vocabulary in `CountVectorsFeaturizer` is not supported during "
                        "incremental training since it requires dynamically adjusting "
                        "layers that correspond to label attributes such as "
                        "`{INTENT_RESPONSE_KEY}`, `{INTENT}`, etc. This is currently not "
                        "possible. In order to avoid this exception we suggest to "
                        "set `use_shared_vocab=False` or train from scratch."